### ISSUE TITLE
/init seems to crash when resume from S3 sleep.

### ISSUE THREAD

**Issue by shigenobuokamoto** [microsoft/WSL#11806]
### Windows Version

Microsoft Windows [Version 10.0.26100.1150]

### WSL Version

2.3.11.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

6.6.36.3-1

### Distro Version

Ubuntu 24.04

### Repro Steps

Windows into S3 sleep.
resume Windows.

### Expected Behavior

return to previous state before sleep.

### Actual Behavior

the WSL Distro crashes.
but, kernel is running.
```
[   30.465930] WSL (1) ERROR: ValidateMessageHeader:329: Protocol error: Received message size:
[   30.465945] 28, type: 7, sequence: 0. Expected type: 0, expected sequence: 2, channel: init
[   30.465972] WSL (2) ERROR: ValidateMessageHeader:329: Protocol error: Received message size:
[   30.476375] hv_utils: TimeSync IC version 4.0
[   30.485262] libc++abi:
[   30.485271] terminating due to uncaught exception of type wil::ResultException: Invalid argument @C:/__w/1/s/src/shared/inc\SocketChannel.h:331 (ValidateMessageHeader)
```

**Comment by g2flyer** [comment]
i see same behavior, also besides sleep (not surprisingly) also with hibernate: distros die (both `wsl` and `wsl --system`) but kernel (as of `wsl --debug-shell`) survives ...

**Comment by rowleya** [comment]
I also get this happening now.  It also seemed to happen randomly when I walked away and came back to my laptop...

**Comment by rowleya** [comment]
Here are my logs after just getting this.  I ran the log collector just before hibernate, and then came back from hibernate and it had crashed.  Hope it helps!
[WslLogs-2024-07-19_07-15-38.zip](https://github.com/user-attachments/files/16303025/WslLogs-2024-07-19_07-15-38.zip)

**Comment by benhillis** [comment]
Thanks for reporting and all the logs above, @oneblue this looks related to your socket protocol changes (or maybe those changes are catching something misbehaving). Could you please take a look?

**Comment by flyingllama87** [comment]
Same issue for me.

**Comment by jamez70** [comment]
I'm having the same issue

**Comment by benhillis** [comment]
We have a fix for this we will be rolling out soon.

**Comment by shigenobuokamoto** [comment]
i have confirmed that this has been fixed in [2.3.12](https://github.com/microsoft/WSL/releases/tag/2.3.12).

thanks so much WSL team.

**Comment by heartacker** [comment]
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x8007019e
Error: 0x8007019e ??????? Linux ? Windows ????

Press any key to continue...

**Comment by jamez70** [comment]
I  have installed 2.3.12 without any issues. The problem usually occurred overnight for me so I will see tomorrow morning if the terminals are still open.

**Comment by pmyers** [comment]
Updated to 2.3.12 via  wsl --update --pre-release and I did not see the WSL2 guest OS crash (in my case Fedora) following a resume from S3. Thank you for the quick fix on this one 👍

**Comment by Pvtsameerkhaan** [comment]
wsl problem not solve

**Comment by jamez70** [comment]
Solved for me. Hasn't crashed since

**Comment by rowleya** [comment]
Solved for me too since the 2.3.12 update (not tried the one just released yet)
